eureka-core/src/main/java/com/netflix/eureka/resources/InstanceResource.java

```diff
@@ -277,16 +277,22 @@ public Response updateMetadata(@Context UriInfo uriInfo) {
     @DELETE
     public Response cancelLease(
             @HeaderParam(PeerEurekaNode.HEADER_REPLICATION) String isReplication) {
-        boolean isSuccess = registry.cancel(app.getName(), id,
+        try {
+            boolean isSuccess = registry.cancel(app.getName(), id,
                 "true".equals(isReplication));
 
-        if (isSuccess) {
-            logger.debug("Found (Cancel): {} - {}", app.getName(), id);
-            return Response.ok().build();
-        } else {
-            logger.info("Not Found (Cancel): {} - {}", app.getName(), id);
-            return Response.status(Status.NOT_FOUND).build();
+            if (isSuccess) {
+                logger.debug("Found (Cancel): {} - {}", app.getName(), id);
+                return Response.ok().build();
+            } else {
+                logger.info("Not Found (Cancel): {} - {}", app.getName(), id);
+                return Response.status(Status.NOT_FOUND).build();
+            }
+        } catch (Throwable e) {
```

Comment:
I know this is already merged, but it isn't released yet.

This should be catching `Exception`, there are very few scenarios in which `Throwable` should be caught and even worse here, swallowed.